pyrokinetics/equilibrium/equilibrium.py

```diff
@@ -321,11 +321,17 @@ def __init__(
         R_major = np.asfarray(R_major) * eq_units["len"]
         r_minor = np.asfarray(r_minor) * eq_units["len"]
         Z_mid = np.asfarray(Z_mid) * eq_units["len"]
+
+        B_0 = B_0 * cocos_factors["BT"] * eq_units["B"]
+        I_p = I_p * cocos_factors["IP"] * eq_units["I"]
+
+        Ip_sign = int(np.sign(I_p))
+
         # Ensure psi is 1D and monotonically increasing
         if len(psi.shape) != 1:
             raise ValueError("The grid psi must be 1D.")
-        if np.any(np.diff(psi) <= 0):
-            raise ValueError("The grid psi must have a positive spacing.")
+        if np.any(np.diff(psi * Ip_sign) <= 0):
+            raise ValueError(f"The grid {Ip_sign} * psi must have a positive spacing.")
```

Comment:
It'd be better to not have an f-string and just say something like `"The grid 'sign(I_p) * psi' must have a positive spacing."`, as otherwise the error message would change between different files and the user might not know why.